### PR TITLE
Create release note and add it to the release automatically

### PR DESCRIPTION
**Pull request by PooyaKharamesh**
Using the functionalities in github to create a release note based on the changes from that last release until the new release and add them to the release 

If any of the steps in generating the release notes fails then an empty release note would be added and partners they can update the notes form the release page on the repo

### CODE REVIEW

File: Actions/CreateReleaseNotes/CreateReleaseNotes.ps1 [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    if ([bool]($latestRelease.PSobject.Properties.name -match "tag_name")) {
+        $latestReleaseTag = $latestRelease.tag_name
+    }
+
+    $realseNotes = GetReleaseNotes -token $token -api_url $ENV:GITHUB_API_URL -repository $ENV:GITHUB_REPOSITORY  -tag_name $tag_name -previous_tag_name $latestReleaseTag
+
+    Write-Host "set-output name=realseNotes::$realseNotes"
+}
+catch {
+    Write-Host "::Error:: Couldn't create release notes. Error was $($_.Exception.Message)"
```
[the remark this answers is not in the document]
Comment (freddydk) [reply Actions/CreateReleaseNotes/CreateReleaseNotes.ps1:27]: No. https://github.com/microsoft/AL-Go/blob/bfaf8b9b21a7f68ffba7372cfbf628789822ef4c/Actions/AL-Go-Helper.ps1#L71 - OutputError outputs the error and sets the return value from PowerShell to 1 (when it exits sometime), meaning that GitHub should see this as a failed build.

File: Actions/CreateReleaseNotes/CreateReleaseNotes.ps1 [hunk 2]
```diff
@@ -0,0 +1,34 @@
+Param(
+    [string] $actor,
+    [string] $token,
```
[the remark this answers is not in the document]
Comment (freddydk) [reply Actions/CreateReleaseNotes/CreateReleaseNotes.ps1:3]: Let's just get this one in, then I will merge mine after this.

**Comment by freddydk** [comment]
Very nice that you found the Mock function in Pester - that can definitely help when writing tests against actions.

File: Actions/CreateReleaseNotes/CreateReleaseNotes.ps1 (hunk 2 above)
[the remark this answers is not in the document]
Comment (PooyaKharamesh) [reply Actions/CreateReleaseNotes/CreateReleaseNotes.ps1:3]: resolving per comment

File: Actions/CreateReleaseNotes/CreateReleaseNotes.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (PooyaKharamesh) [reply Actions/CreateReleaseNotes/CreateReleaseNotes.ps1]: sure

**Pull request merged**